aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu

```diff
@@ -49,68 +51,74 @@ namespace {
   }
 }
 
+void s_addmm_out_csr_sparse_dense_cuda_worker(int64_t nnz, int64_t m, int64_t n, int64_t k, Tensor& r_, const Scalar& beta, const Tensor& t, const Scalar& alpha, Tensor& crow_indices, Tensor& col_indices, Tensor& values, const Tensor& dense) {
+  TORCH_INTERNAL_ASSERT(nnz > 0);
+
+  // No half support, so we don't have to use CUDATypeConversion
+  AT_DISPATCH_FLOATING_AND_COMPLEX_TYPES(
+      values.scalar_type(), "addmm_sparse_cuda", [&] {
+        scalar_t cast_beta = beta.to<scalar_t>();
+        scalar_t cast_alpha = alpha.to<scalar_t>();
+        Tensor r__;
+        if (cast_beta == scalar_t(0)) {
+          r_.zero_();
+        } else if (!is_same_tensor(t, r_)) {
+          r_.copy_(t);
+        }
+        if(r_.stride(0) == 1 && r_.stride(1) == r_.size(0)) {
+          r__ = r_;
+        } else {
+          // Note: This storage arrangement is preferred due to most of the CUDA kernels handle only contiguous tensors
+          r__ = r_.transpose(0, 1).clone(at::MemoryFormat::Contiguous);
+          r__.transpose_(0, 1);
+        }
+        Tensor dense_;
+        char transpose_dense;
+        if(dense.stride(0) == 1 && dense.stride(1) == dense.size(0)) {
```

Comment:
given that cusparse APIs support setting ldb, this seems like a too strict condition? dense.stride(1) can be larger than dense.size(0)

Reply:
Thanks @ngimel , currently `ldb` is set in this way: https://github.com/pytorch/pytorch/blob/1bd22e28b318b15cbfe37ee1a112df9423aab328/aten/src/ATen/native/sparse/cuda/SparseCUDATensorMath.cu#L106

If I understand well the first stride `stride(1)` can be `size(0) * s` where `s` is some positive integer.  I could compute `s`  to consider this case, however  I would like to test this too, but I'm not sure how can I generate this test case.  cc @pearu

Reply:
Indeed, given that csrmm2 supports strided inputs, forcing `dense` to be contiguous is unnecessary.
In fact, the same applies to `r`.

@ngimel, given that @aocsa has only 3 days left to work on this ghstack, would it be ok to land this PR provided that @aocsa will create the corresponding issues that we'll address in follow-up PRs?

Reply:
cc @bhosmer

Reply:
Sure, we can file an issue about relaxing constraints and not pursue it here. I would like to better understand what's going on with device checking though.